src/account/Settings.tsx

```diff
@@ -327,10 +328,17 @@ export class Account extends React.Component<Props, State> {
     this.setState({ showAccountKeyModal: false })
   }
 
-  onPressContinueWithAccountRemoval = () => {
-    ValoraAnalytics.track(SettingsEvents.start_account_removal)
-    this.setState({ showAccountKeyModal: false })
-    this.props.navigation.navigate(Screens.BackupPhrase, { navigatedFromSettings: true })
+  onPressContinueWithAccountRemoval = async () => {
+    try {
+      this.setState({ showAccountKeyModal: false })
+      const pinIsCorrect = await ensurePincode()
+      if (pinIsCorrect) {
```

Comment:
Is there a reason this new feature was added specifically in this PR?

Reply:
Was having an issue where the recovery phrase would display as empty, but this also occurs in main it's own PR makes sense.

https://user-images.githubusercontent.com/26950305/199777055-3f4cb527-9916-4c5d-a429-53b5ce4947cc.mov

Reply:
i think if the issue is already there in main, we should be raising a separate PR to address it rather than sneaking it in here. it's important that PR's are focused, for many reasons but a few that come to mind - focused PR's serves as better documentation (e.g. if you see a line of code that you don't understand, you can see which PR introduced it to get more context) and allows us to detect which features caused bugs more easily (e.g. rewinding commits until the issue cannot be repro'ed anymore).

Reply:
The reason I fixed it hear instead of main is it was causing issues with the e2e tests passing. On main they pass, but after the upgrades they fail on this branch.